shared/src/business/entities/Document.test.js

```diff
@@ -205,6 +214,444 @@ describe('Document entity', () => {
+            { ...mockDocumentData, documentType: 'Petition' },
+            { applicationContext },
+          );
+
+          expect(internalDocument.isValid()).toBeTruthy();
+        });
+      });
+
+      describe('documentType is in the list of documents that require filedBy', () => {
+        // when documentType
+        describe('external filing events', () => {
+          describe('that are not autogenerated', () => {
+            it('should fail validation when "filedBy" is not provided', () => {
+              const document = new Document(
+                {
+                  ...A_VALID_DOCUMENT,
+                  documentId: '777afd4b-1408-4211-a80e-3e897999861a',
+                  documentType: EXTERNAL_DOCUMENT_TYPES[0],
+                  eventCode: 'TRAN',
+                  filedBy: undefined,
+                  isOrder: true,
+                  secondaryDate: '2019-03-01T21:40:46.415Z',
+                },
+                { applicationContext },
+              );
+              expect(document.isValid()).toBeFalsy();
+              expect(document.filedBy).toBeUndefined();
+            });
+
+            it('should pass validation when "filedBy" is provided', () => {
+              const document = new Document(
+                {
+                  ...A_VALID_DOCUMENT,
+                  documentId: '777afd4b-1408-4211-a80e-3e897999861a',
+                  documentType: EXTERNAL_DOCUMENT_TYPES[0],
+                  eventCode: 'TRAN',
+                  filedBy: 'Test Petitioner1',
+                  isOrder: true,
+                  secondaryDate: '2019-03-01T21:40:46.415Z',
+                },
+                { applicationContext },
+              );
+
+              expect(document.isValid()).toBeTruthy();
+            });
+          });
+
+          describe('that are autogenerated', () => {
+            // is: joi.string().valid('Notice of Change of Address'),
+            it('should pass validation when "isAutoGenerated" is true and "filedBy" is undefined', () => {
+              // then: joi.when('isAutoGenerated'  is: false,
+              const documentWithoutFiledBy = omit(A_VALID_DOCUMENT, 'filedBy');
+              const document = new Document(
+                {
+                  ...documentWithoutFiledBy,
+                  documentId: '777afd4b-1408-4211-a80e-3e897999861a',
+                  documentType: 'Notice of Change of Address',
+                  eventCode: 'NCA',
+                  isAutoGenerated: true,
+                  isOrder: true,
+                  secondaryDate: '2019-03-01T21:40:46.415Z',
+                },
+                { applicationContext },
+              );
+
+              expect(document.filedBy).toBeUndefined();
+              expect(document.isValid()).toBeTruthy();
+            });
+
+            it('should pass validation when "isAutoGenerated" is undefined and "filedBy" is undefined', () => {
+              // then: joi.when('isAutoGenerated'  is: false,
```

Comment:
I'm not clear on the intention behind this type of comment - what's its purpose? I see it a few times in this test file.

Reply:
These were notes from building the tests, I'll remove them.

Reply:
Updated!